charts/tango-example/templates/tango-rest.yaml

```diff
@@ -0,0 +1,86 @@
+{{ if .Values.tangorest.enabled }}
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: tango-rest-{{ template "tango-example.name" . }}-{{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: tango-rest-{{ template "tango-example.name" . }}
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+    helm.sh/chart: "{{ template "tango-example.chart" . }}"
+spec:
+  type: ClusterIP
+  ports:
+  - name: rest
+    protocol: TCP
+    port: 80
+    targetPort: rest
+  selector:
+    app.kubernetes.io/name: tango-rest-{{ template "tango-example.name" . }}
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: tango-rest-{{ template "tango-example.name" . }}-{{ .Release.Name }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: tango-rest-{{ template "tango-example.name" . }}
+    app.kubernetes.io/instance: "{{ .Release.Name }}"
+    app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+    helm.sh/chart: "{{ template "tango-example.chart" . }}"
+spec:
+  replicas: {{ .Values.tangorest.replicas }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: tango-rest-{{ template "tango-example.name" . }}
+        app.kubernetes.io/instance: "{{ .Release.Name }}"
+        app.kubernetes.io/managed-by: "{{ .Release.Service }}"
+        helm.sh/chart: "{{ template "tango-example.chart" . }}"
+    spec:
+      {{- if .Values.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+      - name: tango-rest
+        image: "{{ .Values.tangorest.image.registry }}/{{ .Values.tangorest.image.image }}:{{ .Values.tangorest.image.tag }}"
+        imagePullPolicy: {{ .Values.tangorest.image.pullPolicy }}
+        command:
+          - /usr/local/bin/wait-for-it.sh
+          - databaseds-{{ template "tango-example.name" . }}-{{ .Release.Name }}:10000
+          - --timeout=30
+          - --strict
+          - --
+          - /usr/bin/supervisord
```

Comment:
If supervisord is used, how are responsibilities different from K8s orchestration? Shall we settle on just one of the two for managing services as OS processes or running containers?

Reply:
Good point - I forgot about supervisord being there and IMO we should not be doing this kind of thing - pushed up a change for this.